updates to hubs document

anurse opened the pull request:
fixes #9497 as well

Some updates that came up as I was writing #9455 but didn't fit there.

[Internal Review Link](https://review.docs.microsoft.com/en-us/aspnet/core/signalr/hubs?view=aspnetcore-2.1&branch=pr-en-us-9501)

BrennanConroy commented on aspnetcore/signalr/hubs.md:
```diff
@@ -91,9 +91,13 @@ Each property or method in the preceding tables returns an object with a `SendAs
-[!code-csharp[Send messages](hubs/sample/hubs/chathub.cs?range=15-24)]
+* `SendMessage` sends a message to all connected clients, using `Clients.All`.
+* `SendMessageToCaller` sends a message back to caller, using `Clients.Caller`.
+* `SendMessageToGroups` sends a message to all clients in the `SignalR Users` group.
```
```suggestion
* `SendMessageToGroups` sends a message to all clients in the `SignalR Users` group, using `Clients.Group`.
```

BrennanConroy commented on aspnetcore/signalr/hubs.md:
```diff
@@ -111,18 +115,43 @@ Using `Hub<IChatClient>` enables compile-time checking of the client methods. Th
+
+If you have an exceptional condition you *do* want to propagate to the client, you can use the `HubException` class. If you throw a `HubException` from your hub method, SignalR **will** send the entire message to the client, unmodified.
+
+[!code-csharp[ThrowHubException](hubs/sample/hubs/chathub.cs?name=ThrowHubException&highlight=3)]
```
code snippet isn't regioning correctly

anurse commented:
:up: :date: I had to move one of the snippets to be an inline snippet because when I added the `#region` markers it made the "summary" sample at the top of the page ugly. I think the regions are really useful and that initial sample is just there to show a very basic hub, so I think moving it inline works well.

Also addressed @tdykstra 's feedback.

scottaddie commented on aspnetcore/signalr/hubs.md:
```diff
@@ -5,7 +5,7 @@ description: Learn how to use hubs in ASP.NET Core SignalR.
 ms.author: tdykstra
 ms.custom: mvc
-ms.date: 09/12/2018
+ms.date: 11/7/2018
```
```suggestion
ms.date: 11/07/2018
```

scottaddie commented on aspnetcore/signalr/hubs.md:
```diff
@@ -111,18 +123,43 @@ Using `Hub<IChatClient>` enables compile-time checking of the client methods. Th
+[!code-csharp[ThrowHubException](hubs/sample/hubs/chathub.cs?name=ThrowHubException&highlight=3)]
+
+> [!NOTE]
+> SignalR only sends the `Message` property of the exception to the client. The stack trace and other properties on the exception are not available to the client.
```
are not --> aren't

BrennanConroy commented on aspnetcore/signalr/hubs/sample/Hubs/ChatHub.cs:
```diff
@@ -7,32 +7,52 @@ namespace SignalRChat.Hubs
+        #endregion
+
+        #region ThrowHubException
+        public async Task ThrowException()
```
nit: remove async, this wouldn't compile as is.

anurse replied:
It'll compile, but with a warning (CS1998: This async method lacks 'await' operators and will run synchronously...). I originally intended to model an "async" exception here and using `async` ensures that a state-machine is used. We handle synchronous exceptions too though so I think I will switch it up to drop `async`.

anurse commented:
Feedback should be all addressed. @scottaddie, @tdykstra let me know if there's anything else or if this is ready to go! Thanks for the feedback all!

tdykstra commented:
LGTM

scottaddie commented:
Thank you for another awesome PR, @anurse!
